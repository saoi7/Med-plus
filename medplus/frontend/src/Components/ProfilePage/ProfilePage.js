import React from 'react';
import { NavBar } from '../NavBar/NavBar';
import { TextInput } from '../TextInput/TextInput';
import { SubmitInput } from '../SubmitInput/SubmitInput';

// TODO code duplication from AddMedPage
function ProfilePage() {
    return (
        <div className="background-with-logo-image add-med-layout">
            <div className="title font-large">
                My profile
            </div>
            <form className="add-med-form">
                <TextInput labelText="Name: " />
                <TextInput labelText="Email: " />
                <TextInput labelText="Emergency Contact: " />
                <TextInput labelText="Password: " />
                <SubmitInput labelText="Submit user information" type="submit" />
            </form>
            <NavBar />
        </div>
    );
}

export { ProfilePage };