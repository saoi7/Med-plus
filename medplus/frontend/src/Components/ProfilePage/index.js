import React from 'react';
import NavBar from '../NavBar';
import Input from '../Input';
import SubmitInput from '../SubmitInput';
import { withAuthorization } from '../Session';
import { withFirebase } from '../Firebase';
import { withRouter, useHistory } from 'react-router-dom';

class ProfilePageBase extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="background-with-logo-image add-med-layout">
                <div className="title font-large">
                    My profile
                </div>
                <ProfilePageForm firebase={this.props.firebase}/>
                <NavBar />
            </div>
        );
    }
}

class ProfilePageForm extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
       return (
            <form className="add-med-form">
                <Input labelText="Name" />
                <Input labelText="Email" />
                <Input labelText="Emergency Contact" />
                <Input labelText="Password" />
                <SubmitInput labelText="Submit user information" type="submit" />
            </form>
       );
    }
}


const condition = authUser => !!authUser;
const ProfilePage = withRouter(withFirebase(ProfilePageBase));
export default withAuthorization(condition)(ProfilePage);

export { ProfilePageForm, ProfilePageBase };
