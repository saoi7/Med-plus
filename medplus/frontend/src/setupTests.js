// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { unmountComponentAtNode } from "react-dom";

let test_dom_container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  test_dom_container = document.createElement("div");
  document.body.appendChild(test_dom_container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(test_dom_container);
  test_dom_container.remove();
  test_dom_container = null;
});

export { test_dom_container };
