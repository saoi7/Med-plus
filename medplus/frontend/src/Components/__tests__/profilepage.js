import { MOCK_DB_MED_ENTRIES } from './mock_db_med_entries';
import {test_dom_container} from '../../setupTests';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import { getByTestId, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { MOCK_FIREBASE_PROP } from './mock_firebase_props';
import flushPromises from 'flush-promises';
import { ProfilePageBase } from '../ProfilePage';

test("snapshot test of ProfilePage", () => {
    const COMPONENT= renderer.create(
        <MemoryRouter>
            <ProfilePageBase firebase={MOCK_FIREBASE_PROP}/>
        </MemoryRouter>
    );
    let tree = COMPONENT.toJSON();
    expect(tree).toMatchSnapshot();
});
