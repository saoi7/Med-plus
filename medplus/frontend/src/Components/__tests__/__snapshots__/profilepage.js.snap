// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of ProfilePage 1`] = `
<div
  className="background-with-logo-image add-med-layout"
>
  <div
    className="title font-large"
  >
    My profile
  </div>
  <form
    className="add-med-form"
  >
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Name: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
      />
    </div>
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Email: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
      />
    </div>
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Emergency Contact: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
      />
    </div>
    <div
      className="add-med-input font-small undefined"
    >
      <label
        className="left-padding"
      >
        Password: 
      </label>
      <input
        className="hidden-text-box left-padding font-small"
      />
    </div>
    <div
      className="add-med-input font-small"
    >
      <input
        className="hidden-submit-button font-small"
        type="submit"
        value="Submit user information"
      />
    </div>
  </form>
  <nav
    className="nav-bar"
  >
    <a
      className="font-medium"
      href="/home"
      onClick={[Function]}
    >
      Reminders
    </a>
    <a
      className="font-medium"
      href="/home"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
        />
      </svg>
    </a>
    <a
      className="font-medium"
      href="/account"
      onClick={[Function]}
    >
      Profile
    </a>
  </nav>
</div>
`;
